src/ethereum/prague/vm/eoa_delegation.py

```diff
@@ -48,6 +50,25 @@ def is_valid_delegation(code: bytes) -> bool:
     return False
 
 
+def get_delegated_code_address(code: bytes) -> Optional[Address]:
+    """
+    Get the address to which the code delegates.
+
+    Parameters
+    ----------
+    code: `bytes`
+        The code to get the address from.
+
+    Returns
+    -------
+    address : `Optional[Address]`
+        The address of the delegated code.
+    """
+    if is_valid_delegation(code):
+        return Address(code[EOA_DELEGATION_MARKER_LENGTH:])
```

Comment:
Is it possible for code to be longer than `EOA_DELEGATION_MARKER_LENGTH + 20`?

Reply:
Not for valid delegations. One of the checks we make is that the code size is `EOA_DELEGATION_MARKER_LENGTH + 20`